pallets/parachain-staking/src/lib.rs

```diff
@@ -489,20 +489,23 @@ pub mod pallet {
 					selected_collators_number: collator_count,
 					total_balance: total_staked,
 				});
+				// record inactive collators
+				weight = weight.saturating_add(Self::mark_collators_as_inactive(round.current));
 				// account for Round write
 				weight = weight.saturating_add(T::DbWeight::get().reads_writes(0, 1));
 			} else {
 				weight = weight.saturating_add(Self::handle_delayed_payouts(round.current));
 			}
 
 			// add on_finalize weight
```

Comment:
We are ignoring the proof size for both `award_points_to_block_author` and `cleanup_inactive_collator_info`, but should be fine since we over-estimate above.